homeassistant/components/switch/tplink.py

```diff
@@ -92,19 +91,20 @@ def update(self):
             if self.smartplug.has_emeter:
                 emeter_readings = self.smartplug.get_emeter_realtime()
 
-                self._emeter_params[ATTR_CURRENT_CONSUMPTION] \
-                    = "%.1f W" % emeter_readings["power"]
-                self._emeter_params[ATTR_TOTAL_CONSUMPTION] \
-                    = "%.2f kW" % emeter_readings["total"]
+                self._emeter_params[ATTR_CURRENT_POWER_W] \
+                    = "{:.2f}".format(emeter_readings["power"])
```

Comment:
@rytilahti The keys on the library object haven't been changed, so I'm not sure what you mean.

Reply:
Ah, indeed, sorry, my bad! I hold that bug report thread with new FWs in my mind too closely and did not read the diff properly.

Reply:
:+1: